src/main/java/com/faforever/moderatorclient/ui/main_window/LadderMapPoolController.java

```diff
@@ -176,40 +235,50 @@ private void bindListViewSelectionToSelectedMapProperty(ListView<MapVersionFX> l
         }));
     }
 
-    private void bindSelectedMapPropertyToAddRemoveButtons(ObservableList<MapVersionFX> mapList, AddBracketController controller) {
+    private void bindSelectedMapPropertyToAddRemoveButtons(ObservableList<MapPoolAssignmentFX> mapList, AddBracketController controller, MapPoolFX bracketPool) {
         selectedMap.addListener((observable, oldValue, newValue) -> {
-            if (newValue == null) return;
+            if (newValue == null) {
+                return;
+            }
             Button addButton = controller.addToBracketButton;
             Button removeButton = controller.removeFromBracketButton;
             // enable add/remove buttons
-            addButton.setDisable(mapList.contains(newValue));
-            removeButton.setDisable(!mapList.contains(newValue));
+            addButton.setDisable(mapList.stream().anyMatch(assignmentFX -> matchingPoolAssignmentPredicate.test(newValue, assignmentFX)));
+            removeButton.setDisable(!addButton.isDisable());
             //bind actions for add and remove buttons
             addButton.setOnAction(event -> {
-                    mapList.add(newValue);
-                    addButton.setDisable(true);
-                    removeButton.setDisable(false);
+                MapPoolAssignmentFX newAssignment = new MapPoolAssignmentFX();
+                newAssignment.setWeight(1);
+                newAssignment.setMapParams(newValue.getMapParams());
+                newAssignment.setMapVersion(newValue.getMapVersion());
+                newAssignment.setMapPool(bracketPool);
```

Comment:
You can use the setter chaining. Each setter returns the entity itself

Reply:
Ah okay I can set it up althought it looks like it is used inconsistently on the Fx objects and I must have copied one that returned null on the setters